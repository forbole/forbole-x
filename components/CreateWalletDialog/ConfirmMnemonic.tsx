import {
  Button,
  DialogActions,
  DialogContent,
  DialogContentText,
  Box,
  Typography,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import MnemonicPhraseInput from '../MnemonicPhraseInput'
import useStyles from './styles'

interface ConfirmMnemonicProps {
  description: string
  onConfirm(mnemonic: string): void
  error: string
}

const ConfirmMnemonic: React.FC<ConfirmMnemonicProps> = ({ onConfirm, error, description }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const [mnemonic, setMnemonic] = React.useState('')

  return (
    <>
      <DialogContent className={classes.dialogContent}>
        <DialogContentText>{description}</DialogContentText>
        <MnemonicPhraseInput mnemonic={mnemonic} onChange={setMnemonic} />
        <Box mt={2} mb={4}>
          <Typography variant="body2" color="error">
            {error}
          </Typography>
        </Box>
      </DialogContent>
      <DialogActions>
        <Box flex={1} display="flex" flexDirection="column" mb={3}>
          <Button
            className={classes.button}
            variant="contained"
            color="primary"
            onClick={() => onConfirm(mnemonic)}
          >
            {t('next')}
          </Button>
        </Box>
      </DialogActions>
    </>
  )
}

export default ConfirmMnemonic
