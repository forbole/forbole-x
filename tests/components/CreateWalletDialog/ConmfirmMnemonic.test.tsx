import React from 'react'
import renderer from 'react-test-renderer'
import ConfirmMnemonic from '../../../components/CreateWalletDialog/ConfirmMnemonic'

const onConfirm = jest.fn()

jest.mock('../../../components/MnemonicPhraseInput', () => (props) => (
  <div id="MnemonicPhraseInput" {...props} />
))

describe('component: CreateWalletDialog - ConfirmMnemonic', () => {
  it('renders default state correctly', () => {
    const component = renderer.create(
      <ConfirmMnemonic description="description" error="" onConfirm={onConfirm} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders error state correctly', () => {
    const component = renderer.create(
      <ConfirmMnemonic description="description" error="error" onConfirm={onConfirm} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls onConfirm with correct params', () => {
    const component = renderer.create(
      <ConfirmMnemonic description="description" error="" onConfirm={onConfirm} />
    )
    renderer.act(() => {
      component.root
        .findByProps({
          id: 'MnemonicPhraseInput',
        })
        .props.onChange('mnemonic')
    })
    renderer.act(() => {
      component.root.findByType('form').props.onSubmit({ preventDefault: jest.fn() })
    })
    expect(onConfirm).toBeCalledWith('mnemonic')
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

afterEach(() => {
  jest.clearAllMocks()
})
