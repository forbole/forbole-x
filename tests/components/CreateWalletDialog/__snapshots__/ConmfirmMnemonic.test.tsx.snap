// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CreateWalletDialog - ConfirmMnemonic calls onConfirm with correct params 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      description
    </p>
    <div
      id="MnemonicPhraseInput"
      mnemonic="mnemonic"
      onChange={[Function]}
    />
    <div
      className="MuiBox-root MuiBox-root-13"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorError"
      >
        
      </p>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-14"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</form>
`;

exports[`component: CreateWalletDialog - ConfirmMnemonic renders default state correctly 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      description
    </p>
    <div
      id="MnemonicPhraseInput"
      mnemonic=""
      onChange={[Function]}
    />
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorError"
      >
        
      </p>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
      </button>
    </div>
  </div>
</form>
`;

exports[`component: CreateWalletDialog - ConfirmMnemonic renders error state correctly 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      description
    </p>
    <div
      id="MnemonicPhraseInput"
      mnemonic=""
      onChange={[Function]}
    />
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorError"
      >
        error
      </p>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-12"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</form>
`;
